Tests/SonarQube.MSBuild.Tasks.IntegrationTests/TargetsTests/RoslynTargetsTests.cs

```diff
@@ -23,24 +23,53 @@ public class RoslynTargetsTests
         #region SetRoslynSettingsTarget tests
 
         [TestMethod]
+        [Description("Checks the happy path i.e. ruleset exists, analyzer assembly exists, no reason to exclude the project")]
+        public void Roslyn_Settings_ValidSetup()
+        {
+            // Arrange
+            BuildLogger logger = new BuildLogger();
+            ProjectRootElement projectRoot = CreateValidProjectSetup(null);
```

Comment:
I re-structured the existing "set properties" tests to try to reduce duplication. "CreateValidProjectSetup" does all of the required folder/file/property set-up to successfully include the SonarLint analyzer.
The non-happy path tests start from the valid state then change one variable e.g. deleting the ruleset.